Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/CapsuleViewportEdit.h

```diff
@@ -18,34 +18,42 @@ namespace AzToolsFramework
     //! It is designed to be usable either by a component mode or by other contexts which are not associated with a
     //! particular component, so it does not contain any reference to an EntityComponentIdPair or other component-based
     //! identifier.
-    class CapsuleViewportEdit
+    class CapsuleViewportEdit : public BaseShapeViewportEdit
     {
     public:
         CapsuleViewportEdit() = default;
         virtual ~CapsuleViewportEdit() = default;
 
-        void SetupCapsuleManipulators(const ManipulatorManagerId manipulatorManagerId);
-        void TeardownCapsuleManipulators();
-        void UpdateCapsuleManipulators();
-        void ResetCapsuleManipulators();
-
-        virtual AZ::Transform GetCapsuleWorldTransform() const = 0;
-        virtual AZ::Transform GetCapsuleLocalTransform() const = 0;
-        virtual AZ::Vector3 GetCapsuleNonUniformScale() const;
-        virtual float GetCapsuleRadius() const = 0;
-        virtual float GetCapsuleHeight() const = 0;
-        virtual void SetCapsuleRadius(float radius) = 0;
-        virtual void SetCapsuleHeight(float height) = 0;
-        virtual void BeginEditing();
-        virtual void FinishEditing();
-
-    protected:
+        void InstallGetRotationOffset(AZStd::function<AZ::Quaternion()> getRotationOffset);
+        void InstallGetCapsuleRadius(AZStd::function<float()> getCapsuleRadius);
+        void InstallGetCapsuleHeight(AZStd::function<float()> getCapsuleHeight);
+        void InstallSetCapsuleRadius(AZStd::function<void(float)> setCapsuleRadius);
+        void InstallSetCapsuleHeight(AZStd::function<void(float)> setCapsuleHeight);
+
+        // BaseShapeViewportEdit overrides ...
+        void Setup(const ManipulatorManagerId manipulatorManagerId = g_mainManipulatorManagerId);
+        void Teardown();
+        void UpdateManipulators();
+        void ResetValues();
+        void AddEntityComponentIdPair(const AZ::EntityComponentIdPair& entityComponentIdPair);
+
         void OnCameraStateChanged(const AzFramework::CameraState& cameraState);
```

Comment:
Missing the `override` at the end of these functions?

Reply:
oh yeah, good catch, thanks!